src/util/simplify_expr_struct.cpp

```diff
@@ -151,12 +151,13 @@ simplify_exprt::simplify_member(const member_exprt &expr)
       if(!offset_int.has_value())
         return unchanged(expr);
 
-      const exprt &struct_offset=op.op1();
+      const exprt &struct_offset = to_byte_extract_expr(op).offset();
       exprt member_offset = from_integer(*offset_int, struct_offset.type());
       plus_exprt final_offset(struct_offset, member_offset);
       simplify_node(final_offset);
 
-      byte_extract_exprt result(op.id(), op.op0(), final_offset, expr.type());
+      byte_extract_exprt result(
+        op.id(), to_byte_extract_expr(op).op(), final_offset, expr.type());
```

Comment:
Maybe we can avoid doing `to_byte_extract_expr` multiple times?

Reply:
done